src/zenml/stack/stack.py

```diff
@@ -660,6 +678,15 @@ def prepare_pipeline_deployment(
                     f"    `zenml stack up`\n"
                 )
 
+        if self.requires_remote_server and Client().zen_store.is_local_store():
+            raise RuntimeError(
+                "Remote orchestrators and step operators require a remote "
+                "ZenML server. To run a pipeline with this stack you need to "
+                "connect to a remote ZenML server first. Check out "
+                "https://docs.zenml.io/getting-started/deploying-zenml for "
+                "more information on how to deploy ZenML."
```

Comment:
Please feel free to reformat in your IDE. My suggestion is more on wording.

```suggestion
                "Stacks with remote components such as remote orchestrators "
                "and step operators require a remote "
                "ZenML server. To run a pipeline with this stack you need to "
                "connect to a remote ZenML server first. Check out "
                "https://docs.zenml.io/getting-started/deploying-zenml for "
                "more information on how to deploy ZenML."
```